src/test/scala/chiselTests/BlackBox.scala

```diff
@@ -192,7 +203,23 @@ class BlackBoxSpec extends ChiselFlatSpec {
       assert(DataMirror.modulePorts(m) == Seq("in" -> m.io.in, "out" -> m.io.out))
     })
   }
-  "A BlackBoxed using suggestName(\"io\")" should "work (but don't do this)" in {
+  "A BlackBox using suggestName(\"io\")" should "work (but don't do this)" in {
     assertTesterPasses({ new BlackBoxTesterSuggestName }, Seq("/chisel3/BlackBoxTest.v"), TesterDriver.verilatorOnly)
   }
+
+  "A BlackBox with no 'val io'" should "give a reasonable error message" in {
+    (the [ChiselException] thrownBy {
+      ChiselStage.elaborate(new Module {
+        val inst = Module(new BlackBoxNoIO)
+      })
+    }).getMessage should include ("must have a port named 'io'")
+  }
+
+  "A BlackBox with non-Record 'val io'" should "give a reasonable error message" in {
+    (the [ChiselException] thrownBy {
+      ChiselStage.elaborate(new Module {
+        val inst = Module(new BlackBoxUIntIO)
+      })
+    }).getMessage should include ("must have a port named 'io'")
```

Comment:
shouldn't this message say "must have a port of type Record named 'io'", since in this example there *is* a port of type IO?

Reply:
It's there later in the message, made more explicit in the test